From david.vrabel@csr.com  Thu Jun 19 11:53:24 2008
From: David Vrabel <david.vrabel@csr.com>
Date: Mon, 09 Jun 2008 16:02:26 +0100
Subject: UWB: fix oops when notifying of BG joining/leaving
To: linux-usb@vger.kernel.org
Cc: Greg KH <greg@kroah.com>
Message-ID: <484D4602.4090901@csr.com>

From: David Vrabel <david.vrabel@csr.com>

There's no need to claim an additional uwb_dev reference in uwb_notify.

Signed-off-by: David Vrabel <david.vrabel@csr.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/uwb/lc-dev.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/drivers/uwb/lc-dev.c
+++ b/drivers/uwb/lc-dev.c
@@ -81,8 +81,7 @@ EXPORT_SYMBOL_GPL(uwb_notifs_deregister)
 /*
  * Notify all event handlers of a given event on @rc
  *
- * We are called with a valid reference to the device. Obtain another
- * reference before handing off to callback, release on return.
+ * We are called with a valid reference to the device.
  */
 void uwb_notify(struct uwb_rc *rc, struct uwb_dev *uwb_dev, enum uwb_notifs event)
 {
@@ -90,12 +89,9 @@ void uwb_notify(struct uwb_rc *rc, struc
 	if (mutex_lock_interruptible(&rc->notifs_chain.mutex))
 		return;
 	if (!list_empty(&rc->notifs_chain.list)) {
-		uwb_dev_get(uwb_dev);
-		list_for_each_entry(handler, &rc->notifs_chain.list,
-				    list_node) {
+		list_for_each_entry(handler, &rc->notifs_chain.list, list_node) {
 			handler->cb(handler->data, uwb_dev, event);
 		}
-		uwb_dev_put(uwb_dev);
 	}
 	mutex_unlock(&rc->notifs_chain.mutex);
 }
